java/flight/flight-sql-jdbc-core/src/main/java/org/apache/arrow/driver/jdbc/ArrowFlightConnection.java

```diff
@@ -110,6 +110,7 @@ private static ArrowFlightSqlClientHandler createNewClientHandler(
     } catch (final SQLException e) {
       try {
         allocator.close();
+        allocator.getChildAllocators().forEach(BufferAllocator::close);
```

Comment:
1) Can we use AutoCloseables?
2) This needs to come before the root allocator is closed anyways